tests/Microsoft.Identity.Test.Unit/ManagedIdentityTests/ManagedIdentityTests.cs

```diff
@@ -1301,5 +1302,109 @@ public async Task MixedUserAndSystemAssignedManagedIdentityTestAsync()
+                        endpoint,
+                        ManagedIdentityTests.Resource,
+                        "",
+                        managedIdentitySource,
+                        statusCode: HttpStatusCode.InternalServerError);
+                }
+
+                MsalServiceException msalException = null;
+                try
+                {
+                    await mi.AcquireTokenForManagedIdentity(ManagedIdentityTests.Resource)
+                        .ExecuteAsync().ConfigureAwait(false);
+                }
+                catch (Exception ex)
+                {
+                    msalException = ex as MsalServiceException;
+                }
+                Assert.IsNotNull(msalException);
+
+                // ensure that the first request was made and retried 3 times
+                Assert.AreEqual(LinearRetryPolicy.numRetries, ManagedIdentityRequest.DEFAULT_MANAGED_IDENTITY_MAX_RETRIES);
+                Assert.AreEqual(httpManager.QueueSize, 0);
+
+                for (int i = 0; i < NUM_500; i++)
+                {
+                    httpManager.AddManagedIdentityMockHandler(
+                        endpoint,
+                        ManagedIdentityTests.Resource,
+                        "",
+                        managedIdentitySource,
+                        statusCode: HttpStatusCode.InternalServerError);
+                }
+
+                msalException = null;
+                try
+                {
+                    await mi.AcquireTokenForManagedIdentity(ManagedIdentityTests.Resource)
+                        .ExecuteAsync().ConfigureAwait(false);
+                }
+                catch (Exception ex)
+                {
+                    msalException = ex as MsalServiceException;
+                }
+                Assert.IsNotNull(msalException);
+
+                // ensure that the second request was made and retried 3 times
+                // (numRetries would be x2 if retry policy was NOT per request)
+                Assert.AreEqual(LinearRetryPolicy.numRetries, ManagedIdentityRequest.DEFAULT_MANAGED_IDENTITY_MAX_RETRIES);
+                Assert.AreEqual(httpManager.QueueSize, 0);
+
+                for (int i = 0; i < NUM_500; i++)
+                {
+                    httpManager.AddManagedIdentityMockHandler(
+                        endpoint,
+                        ManagedIdentityTests.Resource,
+                        "",
+                        managedIdentitySource,
+                        statusCode: HttpStatusCode.InternalServerError);
+                }
+
+                msalException = null;
+                try
+                {
+                    await mi.AcquireTokenForManagedIdentity(ManagedIdentityTests.Resource)
+                        .ExecuteAsync().ConfigureAwait(false);
+                }
+                catch (Exception ex)
+                {
+                    msalException = ex as MsalServiceException;
+                }
```

Comment:
This can be rewritten as:
```suggestion
                msalException = await Assert.ThrowsExceptionAsync<MsalServiceException>( async() =>
                    await mi.AcquireTokenForManagedIdentity(ManagedIdentityTests.Resource)
                        .ExecuteAsync().ConfigureAwait(false));               
```

Reply:
Is it better to do it the way you suggested? And why? I like the idea of a try/catch as that's what we do in the msal-node tests. Can you explain why I should change it?

Reply:
Using Assert.ThrowsExceptionAsync<MsalServiceException>() makes the intent of the test clear at a glance. A try/ catch is still perfectly valid, as long as you assert inside the catch block. There’s also a third option:

```cs
[TestMethod]
[ExpectedException(typeof(MsalServiceException))]
public async Task AcquireToken_ThrowsAsync()
{
    await mi.AcquireTokenForManagedIdentity(Resource)
            .ExecuteAsync();
}
```

Reply:
done